core/src/test/java/io/grpc/util/UtilServerInterceptorsTest.java

```diff
@@ -111,7 +116,29 @@ public void onReady() {
     getSoleMethod(intercepted).getServerCallHandler().startCall(call, headers).onComplete();
     getSoleMethod(intercepted).getServerCallHandler().startCall(call, headers).onHalfClose();
     getSoleMethod(intercepted).getServerCallHandler().startCall(call, headers).onReady();
-    assertEquals(5, call.numCloses);
+    ServerCall.Listener<Void> callDoubleSreListener =
```

Comment:
This test method is starting to test a lot of different conditions. I would prefer if this new test code is added as two new test cases.